pandas/tests/test_index.py

```diff
@@ -2447,6 +2443,567 @@ def test_slice_keep_name(self):
         idx = Int64Index([1, 2], name='asdf')
         self.assertEqual(idx.name, idx[1:].name)
 
+class TestRangeIndex(Numeric, tm.TestCase):
+    _holder = RangeIndex
+    ### what does the following do?
+    #_multiprocess_can_split_ = True
+
+    def setUp(self):
+        self.indices = dict(index = RangeIndex(0, 20, 2))
+        self.setup_indices()
+
+    def create_index(self):
+        return RangeIndex(5)
+
+    def test_too_many_names(self):
+        def testit():
+            self.index.names = ["roger", "harold"]
+        assertRaisesRegexp(ValueError, "^Length", testit)
+
+    def test_constructor(self):
+        index = RangeIndex(5)
+        expected = np.arange(5, dtype=np.int64)
+        self.assert_numpy_array_equal(index, expected)
+
+        index = RangeIndex(1, 5)
+        expected = np.arange(1, 5, dtype=np.int64)
+        self.assert_numpy_array_equal(index, expected)
+
+        index = RangeIndex(1, 5, 2)
+        expected = np.arange(1, 5, 2, dtype=np.int64)
```

Comment:
You need a whole bunch of tests that take an ``Index`` and should return a ``RangeIndex`` for certain (e.g. ranges) input, I don't know if these actually exist for ``Int64Index`` (other they would fail).